cms/core/models/base.py

```diff
@@ -1,7 +1,7 @@
+from functools import cached_property
```

Comment:
My understanding is that the Django version is slightly more performant in most standard cases... the native version has a lot more going on in terms of thread-safety, which isn't really relevant to Django model (or view) instances